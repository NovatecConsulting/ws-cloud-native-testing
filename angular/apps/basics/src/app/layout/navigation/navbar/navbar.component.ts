import { Component } from '@angular/core';
import { DEFINED_ROUTES } from '../../../routes';

@Component({
  selector: 'basics-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  routes = DEFINED_ROUTES.filter((route) => !!route.displayName);
}
