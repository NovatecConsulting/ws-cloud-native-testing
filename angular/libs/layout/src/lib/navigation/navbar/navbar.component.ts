import { Component, Input } from '@angular/core';
import { DefinedRoute } from '../model';

@Component({
  selector: 'layout-navbar',
  templateUrl: './navbar.component.html',
  styleUrls: ['./navbar.component.scss'],
})
export class NavbarComponent {
  @Input() routes: DefinedRoute[] = [];
}
