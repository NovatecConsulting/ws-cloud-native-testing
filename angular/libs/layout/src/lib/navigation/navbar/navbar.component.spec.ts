import { NavbarComponent } from './navbar.component';
import { MockBuilder, MockRender } from 'ng-mocks';
import { RouterTestingModule } from '@angular/router/testing';

// TODO: fix me, own module
describe('Navbar', () => {
  beforeEach(() => MockBuilder(NavbarComponent).keep(RouterTestingModule));

  it('should stay consistent', () => {
    const fixture = MockRender(NavbarComponent);
    expect(fixture).toMatchSnapshot();
  });
});
