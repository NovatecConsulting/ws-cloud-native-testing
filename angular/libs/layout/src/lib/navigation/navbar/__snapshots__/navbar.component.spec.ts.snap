// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should stay consistent 1`] = `
<mock-render
  __ngContext__="0"
>
  <layout-navbar>
    <mat-toolbar
      class="w-screen"
      color="primary"
    >
    </mat-toolbar>
  </layout-navbar>
</mock-render>
`;
